lib/src/api/opt/capabilities.rs

```diff
@@ -26,43 +57,368 @@
 /// Create a new instance, and allow all capabilities
 #[cfg_attr(feature = "kv-rocksdb", doc = "```no_run")]
 #[cfg_attr(not(feature = "kv-rocksdb"), doc = "```ignore")]
-/// # use surrealdb::opt::capabilities::Capabilities;
+/// # use surrealdb::opt::capabilities::CapabilitiesBuilder;
 /// # use surrealdb::opt::Config;
 /// # use surrealdb::Surreal;
 /// # use surrealdb::engine::local::File;
 /// # #[tokio::main]
 /// # async fn main() -> surrealdb::Result<()> {
-/// let capabilities = Capabilities::all();
+/// let capabilities = CapabilitiesBuilder::all();
 /// let config = Config::default().capabilities(capabilities);
 /// let db = Surreal::new::<File>(("temp.db", config)).await?;
 /// # Ok(())
 /// # }
 /// ```
-///
 /// Create a new instance, and allow certain functions
 #[cfg_attr(feature = "kv-rocksdb", doc = "```no_run")]
 #[cfg_attr(not(feature = "kv-rocksdb"), doc = "```ignore")]
 /// # use std::str::FromStr;
 /// # use surrealdb::engine::local::File;
-/// # use surrealdb::opt::capabilities::Capabilities;
-/// # use surrealdb::opt::capabilities::FuncTarget;
-/// # use surrealdb::opt::capabilities::Targets;
+/// # use surrealdb::opt::capabilities::CapabilitiesBuilder;
 /// # use surrealdb::opt::Config;
 /// # use surrealdb::Surreal;
 /// # #[tokio::main]
 /// # async fn main() -> surrealdb::Result<()> {
-/// let capabilities = Capabilities::default()
-///     .with_functions(Targets::<FuncTarget>::All)
-///     .without_functions(Targets::<FuncTarget>::Some(
-///         [FuncTarget::from_str("http::*").unwrap()].into(),
-///     ));
+/// let capabilities = CapabilitiesBuilder::default()
+///     .with_deny_function("http::*")?;
 /// let config = Config::default().capabilities(capabilities);
 /// let db = Surreal::new::<File>(("temp.db", config)).await?;
 /// # Ok(())
 /// # }
 /// ```
-pub use crate::dbs::Capabilities;
+///
+#[derive(Debug, Clone)]
+pub struct CapabilitiesBuilder {
+	cap: Capabilities,
+	allow_funcs: Targets<FuncTarget>,
+	deny_funcs: Targets<FuncTarget>,
+	allow_net: Targets<NetTarget>,
+	deny_net: Targets<NetTarget>,
+}
+
+impl Default for CapabilitiesBuilder {
+	fn default() -> Self {
+		Self::new()
+	}
+}
+
+impl CapabilitiesBuilder {
+	/// Create a builder with default capabilities enabled.
+	///
+	/// Default capabilities enables live query notifications and all (non-scripting) functions.
+	pub fn new() -> Self {
+		CapabilitiesBuilder {
+			cap: Capabilities::default(),
+			allow_funcs: Targets::All,
```

Comment:
That is not the default that we had documented, but in practice I see that this is already the default. Do we want to allow functions by default? I think specially the `http` class of functions can be quite dangerous in many environments where SurrealDB can be running. By default, any SurrealDB user would be able to make network requests on the local network where SurrealDB is running. I would probably just disable functions by default as we already describe in the documentation. If you want we can discuss this separately,

Reply:
I will resolve this comment and make a separate PR to propose this change.